### PR TITLE
[BAEL-7632] Convert a phone number in words to number with Java

### PR DESCRIPTION
**Pull request by mehvishashiq**
mehvishashiq155@gmail.com

### CODE REVIEW

File: core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        assertThrows(IllegalArgumentException.class, () -> {
+            UseHashMapToConvertPhoneNumberInWordsToNumber
+                    .convertPhoneNumberInWordToNumber("five eight invalid two four null eight");
+        });
+
+    }
+
+
+    @Test
+    public void givenNothing_WhenGetWordAsMultiplier_ThenEquivalentNumber() {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java:33]: `givenNothing` doesn't add much. We can remove it from the test name.

File: core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import org.junit.jupiter.api.Test;
+import java.util.Map;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class UseHashMapToConvertPhoneNumberInWordToNumberUnitTest {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java:5]: JUnit 5 classes don't need `public` on the `class` or the methods

File: core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+
+    @Test
+    public void givenNothing_WhenGetWordAsMultiplier_ThenEquivalentNumber() {
+        Map<String, Integer> result = UseHashMapToConvertPhoneNumberInWordsToNumber.getWordAsMultiplier();
+        assertEquals(2, result.get("double"));
+        assertEquals(null, result.get("invalid"));
+        assertEquals(null, result.get(""));
+        assertEquals(null, result.get("deca"));
+    }
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java:39]: These are very low level tests. Do we need them? We're testing implementation details of the map.

File: core-java-modules/core-java-numbers-7/src/main/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordsToNumber.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+        multiplier.put("double", 2);
+        multiplier.put("triple", 3);
+        multiplier.put("quadruple", 4);
+
+        return multiplier;
+    }
+
+
+    public static Map<String, String> getWordAsDigit() {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-numbers-7/src/main/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordsToNumber.java:46]: We don't need a function that continously builds this map. We need a static constant in the class containing this map. You can do that using `private static` fields and the `static { ... construct here ... }` constructor method.

In higher versions of Java, we can also build a map in an initialiser list using `Map.of`. Or we can use map building code from Guava if we really wanted to.

The map needs to be wrapped by `unmodifiableMap` https://www.geeksforgeeks.org/collections-unmodifiablemap-method-in-java-with-examples/ to make it read only.

Having a method repeatedly make a map is a bad example to set to the reader.

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java (hunk 2 above)
Comment (mehvishashiq) [reply core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java]: Done.

File: core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java (hunk 3 above)
Comment (mehvishashiq) [reply core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java]: Removed the low-level tests.

File: core-java-modules/core-java-numbers-7/src/main/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordsToNumber.java (hunk 4 above)
Comment (mehvishashiq) [reply core-java-modules/core-java-numbers-7/src/main/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordsToNumber.java]: Used the `private static Map.of()` method to fix the issue.

**Comment by mehvishashiq** [comment]
@ashleyfrieze - I've made modifications based on the provided feedback.

File: core-java-modules/core-java-numbers-7/src/main/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordsToNumber.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            "three", "3",
+            "four", "4",
+            "five", "5",
+            "six", "6",
+            "seven", "7",
+            "eight", "8",
+            "nine", "9");
+
+
+    public static String convertPhoneNumberInWordToNumber(String phoneNumberInWord) {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-numbers-7/src/main/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordsToNumber.java:19]: `InWords` not `InWord`

File: core-java-modules/core-java-numbers-7/src/main/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordsToNumber.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                }
+                currentMultiplier = multiplier;
+            } else {
+                String digit = digits.get(word);
+                if (digit != null) {
+                    output.append(digit.repeat(currentMultiplier != null ? currentMultiplier : 1));
+                    currentMultiplier = null;
+                } else {
+                    throw new IllegalArgumentException("Invalid word: " + word);
+                }
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-numbers-7/src/main/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordsToNumber.java:39]: I suggest inverting the `if`.
Then you won't need the `else`

```java
if (digit == null) { 
    throw new IllegalArgumentException(..);
}
// code continues
```

File: core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+class UseHashMapToConvertPhoneNumberInWordToNumberUnitTest {
+
+    @Test
+    void givenStringWithWhiteSpaces_WhenConvertPhoneNumberInWordToNumber_ThenEquivalentNumber() {
+
+        assertEquals("5248888",
+                UseHashMapToConvertPhoneNumberInWordsToNumber
+                        .convertPhoneNumberInWordToNumber("five two four quadruple eight"));
+
+        assertThrows(IllegalArgumentException.class, () -> {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseHashMapToConvertPhoneNumberInWordToNumberUnitTest.java:13]: Each assert is a separate test case here. Let's have multiple test methods.

File: core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseSwitchToConvertPhoneNumberInWordToNumberUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        assertEquals("5248888",
+                UseSwitchToConvertPhoneNumberInWordToNumber
+                        .convertPhoneNumberInWordToNumber("five two four quadruple eight"));
+
+        assertThrows(IllegalArgumentException.class, () -> {
+            UseSwitchToConvertPhoneNumberInWordToNumber
+                    .convertPhoneNumberInWordToNumber("five eight three double triple");
+        });
+
+        assertThrows(IllegalArgumentException.class, () -> {
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseSwitchToConvertPhoneNumberInWordToNumberUnitTest.java:18]: AND HERE.

File: core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseSwitchToConvertPhoneNumberInWordToNumberUnitTest.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        });
+
+    }
+
+    @Test
+    void givenString_WhenGetWordAsMultiplier_ThenEquivalentNumber() {
+        assertEquals(2, UseSwitchToConvertPhoneNumberInWordToNumber
+                .getWordAsMultiplier("double"));
+
+        assertEquals(null, UseSwitchToConvertPhoneNumberInWordToNumber
```
Comment (ashleyfrieze) [inline core-java-modules/core-java-numbers-7/src/test/java/com/baeldung/convertphonenumberinwordstonumber/UseSwitchToConvertPhoneNumberInWordToNumberUnitTest.java:35]: You're testing internals here. No need.

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

**Pull request merged**